### PR TITLE
Order exercises by position

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes https://github.com/exercism/v3-project-management/issues/17

### CODE REVIEW

File: db/migrate/20210402085227_add_position_to_exercises.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  def change
+    add_column :exercises, :position, :integer, null: true, after: :blurb
+
+    Exercise.find_each.with_index do |exercise, position|
+      exercise.update!(position: position)
+    end 
+
+    change_column_null :exercises, :position, false
+
+    add_index :exercises, [:track_id, :type, :position], name: 'index_track_exercise_position'
```
Comment (ErikSchierboom) [inline db/migrate/20210402085227_add_position_to_exercises.rb]: I've added this index to make returning the sorted exercise very fast. As we have a relatively small number of tracks (70-ish) with a relatively small number of exercises (10-200), the total amount will be something like 70*200=14000. I think the cost of the index is probably negligible and the gains more important.

This is the `EXPLAIN` output for `select * from exercises where track_id = 1 order by type, position`:

id? | select_type? | table? | partitions? | type? | possible_keys? | key? | key_len? | ref? | rows? | filtered? | Extra?
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | SIMPLE | exercises | NULL | ref | index_exercises_on_track_id_and_uuid,index_exercises_on_track_id,index_track_exercise_position | index_track_exercise_position | 8 | const | 167 | 100.00 | Using index condition

Comment (ErikSchierboom) [reply db/migrate/20210402085227_add_position_to_exercises.rb]: Update: the `type` has now been removed.

File: db/migrate/20210402085227_add_position_to_exercises.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class AddPositionToExercises < ActiveRecord::Migration[6.1]
+  def change
+    add_column :exercises, :position, :integer, null: true, after: :blurb
```
Comment (iHiD) [inline db/migrate/20210402085227_add_position_to_exercises.rb]:
```suggestion
    add_column :exercises, :position, :integer, null: true, after: :blurb, index: true
```

File: db/migrate/20210402085227_add_position_to_exercises.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+    Exercise.find_each.with_index do |exercise, position|
+      exercise.update!(position: position)
+    end 
+
+    change_column_null :exercises, :position, false
+
+    add_index :exercises, [:track_id, :position]
```
Comment (iHiD) [inline db/migrate/20210402085227_add_position_to_exercises.rb]:
```suggestion
```

**Pull request merged**